[DNM] 404 page

frederickobrien opened the pull request:
Features/tweaks done:

- Change font family of image captions
- External link symbol on artwork stories
- Address bar theme colour
- Flashier platinum award
- Make banner scores out of 30

andrewbridge commented on layouts/default.vue:
```diff
@@ -29,6 +29,7 @@ export default {
     return {
       meta: [
         { hid: "og:url", property: "og:url", content: SITE_URL + this.$route.path },
+        { hid: "theme-color", color: "#292929" },
```
@frederickobrien We actually already have one of these here:

https://github.com/audioxide/website/blob/18233c3c31999dc9b474a09dffadb6c2763a7743/nuxt.config.js#L67

Did Lighthouse complain that it didn't exist?

frederickobrien replied:
Huh, that's odd. It was flagged as an issue here: https://lighthouse-metrics.com, @andrewbridge. Could be a problem with the tool?

andrewbridge replied:
Just checked this, I think it isn't complaining about the `<meta>`. From the help page:

> The audit fails if Lighthouse doesn't find a theme-color meta tag in the page's HTML **and a `theme_color` property in the web app manifest.**

Emphasis is mine, I think we need to add the value to our web app manifest. This is something that is at least partially managed by Nuxt, so we might need to dig into their documentation to work out how to do this correctly.

andrewbridge commented:
This all looks good, would be good to know more about that theme colour meta as per above and then we can work out what's going on. Otherwise this is good to go.

frederickobrien commented:
> 
> 
> This all looks good, would be good to know more about that theme colour meta as per above and then we can work out what's going on. Otherwise this is good to go.

Ok, I've added aria labels to the external link icon so we should be good to go there. I actually set up this PR with a 404 page in mind but I haven't got my head around that yet. Maybe worth waving this through if it looks ok to you and I'll come at that problem afresh later?

frederickobrien commented:
Ok, I stopped umming and ahing and had a go at a custom 404 page for when a URL does not exist. Some success but have of course hit a wall. Hard to tell if this is a Netlify issue or a Vue issue as the bleed between the two here makes the documentation of both less useful than I would like.

What do you reckon, @andrewbridge?

andrewbridge commented:
@frederickobrien with your couple follow up commits following your previous message, you've got there on the Netlify side. If you run the build on your local machine, you'll be able to see the `404/index.html` file that you're now pointing to.

With Nuxt, however, it's a bit. The docs actually suggest using [a layout called `error.vue`](https://nuxtjs.org/docs/2.x/concepts/views#error-page). The one thing I've neglected to do as well is to implement the [`validate` method](https://nuxtjs.org/docs/2.x/components-glossary/pages-validate), which would instruct Nuxt to show the error page if a page or post didn't exist in our data.

I'm a bit torn, because what you've done provides a statically generated 404 page, which I think is pretty important, but it won't play entirely nicely with Nuxt. There may be a middle ground that can give us the benefits of both. I'll can take a look some time this week if you're still hitting a dead end.

frederickobrien commented:
The code is sound, but I still had the old ones locally so had to swap them out. I think that's why the build is failing for now.

andrewbridge commented:
@frederickobrien think it's because in [setup.js](https://github.com/audioxide/website/blob/main/setup.js#L31) we specifically copy files prefixed with `ribbon-`. We could add a check for `badge-` as well.

Incidentally, what's with the lock files in https://github.com/audioxide/website/pull/143/commits/b4fec106ad430aece3df426a9c7fb6244587dfff ?

frederickobrien commented:
> 
> 
> @frederickobrien think it's because in [setup.js](https://github.com/audioxide/website/blob/main/setup.js#L31) we specifically copy files prefixed with `ribbon-`. We could add a check for `badge-` as well.
> 
> Incidentally, what's with the lock files in [b4fec10](https://github.com/audioxide/website/commit/b4fec106ad430aece3df426a9c7fb6244587dfff) ?

I guess it makes sense to have a universal prefix, be it `ribbon-`, `badge-`, `award-`, or whatever else?

Aha, I downloaded some stuff for Scala and it seems to have worked its way into the branch somehow. 'Metals' is the Scala plugin for VSC. Sorry, wasn't sure what to do about that one.

frederickobrien commented:
I've updated setup.js, @andrewbridge, and changed all award files to start with `award-`. Deploy still fails, hits a stumbling block with the new format. Do PRs pull from the main brand repo? If so that may be where the issue is? I've added the new ribbon/sticker files there and kept the old ones for now as not to break the current site. 

I've also done a quick fix for the banner issue Gabs mentioned! Not the most elegant solution but the perfect is the enemy of the good, etc.

frederickobrien commented:
Do you think it'd be worth retracting the 404 changes for now and merging the rest of this, @andrewbridge? Would be nice to get the cosmetic tweaks integrated. Lesson learned also: I'll make my PRs even smaller from now on. Almost nothing to be gained from bundling even two or three things together.

frederickobrien commented:
Ok, I've redone all commits apart from the 404 page and new ribbons as separate PRs. Once the brand repo is rebased and I know the new awards work ok I'll move them over too.

frederickobrien commented:
This will be included in https://github.com/audioxide/website/pull/183
